obj_factory.h

```diff
@@ -102,6 +114,63 @@ class ObjFactory {
         return vec[idx.val].get();
     }
 
+    bool if_exist(const std::string& uri) {
+        if (map.find(uri) == map.end()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    bool erase(const Idx<ObjType>& idx) {
+        if (idx.val >= vec.size()) {
+            return false;
+        }
+        // erase map menber
+        std::string finded_key;
+        const auto* elem_ptr = vec[idx.val].get();
+        for (const auto& elem : map) {
```

Comment:
No need to search `uri` like that. Because uri is inside obj directly. I'll change that